### DISCUSSION TITLE
airflow.providers.microsoft.azure.hooks.wasb get_blobs_list not listing metadata

### DISCUSSION THREAD

**Discussion by lx183** [apache/airflow#32258] [General]
### Apache Airflow version

2.6.2

### What happened

When trying to list blobs in a given container with metadata like last modified date, I only get the blob name

### What you think should happen instead

get_blobs_list should return the blob object with metadata

### How to reproduce

Call airflow.providers.microsoft.azure.hooks.wasb get_blobs_list

### Operating System

NAME="Red Hat Enterprise Linux" VERSION="8.8 (Ootpa)" ID="rhel" ID_LIKE="fedora" VERSION_ID="8.8" PLATFORM_ID="platform:el8" PRETTY_NAME="Red Hat Enterprise Linux 8.8 (Ootpa)" ANSI_COLOR="0;31" CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos" HOME_URL="https://www.redhat.com/" DOCUMENTATION_URL="https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8" BUG_REPORT_URL="https://bugzilla.redhat.com/"  REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8" REDHAT_BUGZILLA_PRODUCT_VERSION=8.8 REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux" REDHAT_SUPPORT_PRODUCT_VERSION="8.8"

### Versions of Apache Airflow Providers

apache-airflow-providers-cncf-kubernetes==7.1.0
apache-airflow-providers-common-sql==1.5.1
apache-airflow-providers-databricks==4.3.0
apache-airflow-providers-ftp==3.4.1
apache-airflow-providers-google==10.2.0
apache-airflow-providers-http==4.4.1
apache-airflow-providers-imap==3.2.1
apache-airflow-providers-microsoft-azure==6.1.2
apache-airflow-providers-sqlite==3.4.1

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Adaverse** [comment]
Yes! It seems it was intended to be this way seeing its usages.
A simple workaround - 
```
container = hook._get_container_client(container_name)
blob_list = container.walk_blobs(...) # or container.list_blobs(...)
# iterate the blob list
```

**Comment by potiuk** [comment]
Converted to a discussion, as indeed, there is no "issue" here. However if someone would like to change the way it works, making a PR for that is always welcome.
